language/move-stdlib/src/natives/vector.rs

```diff
@@ -109,5 +116,26 @@ pub fn native_swap(
     let idx1 = pop_arg!(args, u64) as usize;
     let r = pop_arg!(args, VectorRef);
     let cost = native_gas(context.cost_table(), NativeCostIndex::SWAP, 1);
-    NativeResult::map_partial_vm_result_empty(cost, r.swap(idx1, idx2, &ty_args[0]))
+    NativeResult::map_partial_vm_result_empty(
+        cost,
+        r.swap(idx1, idx2, &ty_args[0])
+            .map_err(native_error_to_abort),
+    )
+}
+
+fn native_error_to_abort(err: PartialVMError) -> PartialVMError {
```

Comment:
This feels a little gross, but we are deprecating the vector natives eventually so probably best to keep the grossness contained to this file

Reply:
I agree. It's a hack and hopefully will go away when we deprecate the vector natives.